File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/Functionality/ValueTextChangePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.ComboBox.Functionality
{
    public partial class ValueTextChangePage : UserControl
    {
        public ValueTextChangePage()
        {
            InitializeComponent();
            //Set DataSource as array of Colors.
            this.mobjComboBox.DataSource = Enum.GetValues(typeof(KnownColor));
        }

        private void mobjComboBox_TextChanged(object sender, EventArgs e)
        {
            //Set text of tested ComboBox as text of indicator TextBox.
            this.mobjComboBoxTextTextBox.Text = this.mobjComboBox.Text;
        }

        private void mobjComboBox_SelectedValueChanged(object sender, EventArgs e)
        {
            if (this.mobjComboBox.SelectedValue != null)
            {
                //Set text of tested ComboBox SelectedValue as text of indicator TextBox.
                this.mobjValueTextBox.Text = this.mobjComboBox.SelectedValue.ToString();
            }
        }

        /// <summary>
        /// Handles the TextChanged event of the mobjTextTextBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjTextTextBox_TextChanged(object sender, EventArgs e)
        {
            if (!string.IsNullOrEmpty(mobjTextTextBox.Text))
            {
                mobjComboBox.Text = mobjTextTextBox.Text;
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/Functionality/ValueTextChangePage.vb
Namespace CompanionKit.Controls.ComboBox.Functionality

	Public Class ValueTextChangePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()
			'Set DataSource as array of Colors.
            Me.mobjComboBox.DataSource = [Enum].GetValues(GetType(Drawing.KnownColor))
        End Sub

        Private Sub mobjComboBox_TextChanged(sender As Object, e As EventArgs) Handles mobjComboBox.TextChanged
            'Set text of tested ComboBox as text of indicator TextBox.
            Me.mobjComboBoxTextTextBox.Text = Me.mobjComboBox.Text
        End Sub

        Private Sub mobjComboBox_SelectedValueChanged(sender As Object, e As EventArgs) Handles mobjComboBox.SelectedValueChanged
            If Me.mobjComboBox.SelectedValue IsNot Nothing Then
                'Set text of tested ComboBox SelectedValue as text of indicator TextBox.
                Me.mobjValueTextBox.Text = Me.mobjComboBox.SelectedValue.ToString()
            End If
        End Sub

        ''' <summary>
        ''' Handles the TextChanged event of the mobjTextTextBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjTextTextBox_TextChanged(sender As Object, e As EventArgs) Handles mobjTextTextBox.TextChanged
            If Not String.IsNullOrEmpty(mobjTextTextBox.Text) Then
                mobjComboBox.Text = mobjTextTextBox.Text
            End If
        End Sub

	End Class

End Namespace